### PR TITLE
[community] Load Models from Sources like `Civitai` into Existing Pipelines

### PR DESCRIPTION
**Pull request by suzukimain**
# What does this PR do?
This PR adds an example that allows you to search for models from `Civitai` and `Hugging Face` by simply entering a keyword and loading models that meet the criteria into existing pipelines.

> Example
```bash
!wget https://raw.githubusercontent.com/suzukimain/auto_diffusers/refs/heads/master/src/auto_diffusers/pipeline_easy.py
```

```python
from pipeline_easy import (
    EasyPipelineForText2Image,
    EasyPipelineForImage2Image,
    EasyPipelineForInpainting,
)

# Search for Civitai
pipeline = EasyPipelineForText2Image.from_civitai(
    "search_word",
    base_model="SD 1.5",
).to("cuda")


# Search for Hugging Face
pipeline = EasyPipelineForText2Image.from_huggingface(
    "search_word",
    checkpoint_format="diffusers",
).to("cuda")
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [X] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by bghira** [comment]
can it use AIR IDs to import them? thanks for this example!

File: examples/model_search/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Searching Models on Civitai and Hugging Face
+Please refer to the original library [here](https://pypi.org/project/auto-diffusers/)
```
Comment (stevhliu) [inline examples/model_search/README.md:2]:
```suggestion
# Search models on Civitai and Hugging Face

The [auto_diffusers](https://github.com/suzukimain/auto_diffusers) library provides additional functionalities to Diffusers such as searching for models on Civitai and the Hugging Face Hub.
Please refer to the original library [here](https://pypi.org/project/auto-diffusers/)
```

File: examples/model_search/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+cd diffusers
+pip install .
+```
+
+Then cd in the example folder and run:
+```bash
+pip install -r requirements.txt
+```
+
+
```
Comment (stevhliu) [inline examples/model_search/README.md:22]:
```suggestion
## Installation

Before running the scripts, make sure to install the library's training dependencies:

> [!IMPORTANT]
> To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the installation up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment.

```bash
git clone https://github.com/huggingface/diffusers
cd diffusers
pip install .
```

Navigate to the example folder and install the `requirements.txt` file.

```bash
pip install -r requirements.txt
```


```

File: examples/model_search/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+##  Example<a name = "Example"></a>
```
Comment (stevhliu) [inline examples/model_search/README.md:23]:
```suggestion
##  Search for models
```

File: examples/model_search/README.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+```python
+# Search for Civitai
+
+from search_for_civitai_and_HF import CivitaiSearchPipeline
+from diffusers import StableDiffusionPipeline
+
+
+model_path = CivitaiSearchPipeline.for_civitai(
+    "any",
```
Comment (stevhliu) [inline examples/model_search/README.md:36]: Would be nice to explicitly specify that `any` refers to `search_word`

File: examples/model_search/README.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+           "stable",
+           checkpoint_format="single_file",
+           download = False
+           )
+
+pipe = StableDIffusionPipeline.from_single_file(model_path).to("cuda")
+```
+
+<a id="Details"></a>
+<details close>
```
Comment (stevhliu) [inline examples/model_search/README.md:71]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Nice, really cool and convenient functionality!

**Comment by suzukimain** [comment]
I was mostly done reflecting the changes.
However, I was not sure how to do these parts.
cc: @stevhliu 

https://github.com/huggingface/diffusers/blob/745d4d06b1cfe5779b618a451475d43cceddbb6c/examples/model_search/README.md?plain=1#L118

https://github.com/huggingface/diffusers/blob/745d4d06b1cfe5779b618a451475d43cceddbb6c/examples/model_search/README.md?plain=1#L148

File: examples/model_search/README.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+Before running the scripts, make sure to install the library's training dependencies:
+
+> [!IMPORTANT]
+> To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the installation up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment.
+
+```bash
+git clone https://github.com/huggingface/diffusers
+cd diffusers
+pip install .
```
Comment (stevhliu) [inline examples/model_search/README.md:16]: Need to close the code snippet here

File: examples/model_search/README.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+```bash
+!wget https://raw.githubusercontent.com/suzukimain/auto_diffusers/refs/heads/master/src/auto_diffusers/pipeline_easy.py
+```
+
+### Search for Civitai
```
Comment (stevhliu) [inline examples/model_search/README.md:21]:
```suggestion
## Load from Civitai
```

File: examples/model_search/README.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+
+# Inpainting
+pipeline = EasyPipelineForInpainting.from_civitai(
+    "search_word",
+    base_model="SD 1.5",
+).to("cuda")
+```
+
+### Search for Hugging Face
```
Comment (stevhliu) [inline examples/model_search/README.md:50]:
```suggestion
## Load from Hugging Face
```

File: examples/model_search/README.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+| search_word     | string, Path           | ー            | The search query string. Can be a keyword, Civitai URL, local directory or file path. |
+| model_type      | string                 | `Checkpoint`  | The type of model to search for.  <br>(for example `Checkpoint`, `TextualInversion`, `Controlnet`, `LORA`, `Hypernetwork`, `AestheticGradient`, `Poses`)      |
+| base_model      | string                 | None          | Trained model tag (for example  `SD 1.5`, `SD 3.5`, `SDXL 1.0`) |
+| torch_dtype     | string, torch.dtype    | None          | Override the default `torch.dtype` and load the model with another dtype.     |
+| force_download  | bool                   | False         | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
+| cache_dir       | string, Path | None    | Path to the folder where cached files are stored. |
+| resume          | bool   | False         | Whether to resume an incomplete download. |
+| token           | string | None          | API token for Civitai authentication. |
```
Comment (stevhliu) [inline examples/model_search/README.md:124]: You can move this to the `## Search Civitai` section

File: examples/model_search/README.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+| search_word           | string, Path        | ー             | The search query string. Can be a keyword, Hugging Face URL, local directory or file path, or a Hugging Face path (`<creator>/<repo>`). |
+| checkpoint_format     | string              | `single_file`  | The format of the model checkpoint.<br>● `single_file` to search for `single file checkpoint` <br>●`diffusers` to search for `multifolder diffusers format checkpoint` |
+| torch_dtype           | string, torch.dtype | None           | Override the default `torch.dtype` and load the model with another dtype. |
+| force_download        | bool                | False          | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
+| cache_dir             | string, Path        | None           | Path to a directory where a downloaded pretrained model configuration is cached if the standard cache is not used.   |
+| token                 | string, bool        | None           | The token to use as HTTP bearer authorization for remote files.  |
```
Comment (stevhliu) [inline examples/model_search/README.md:152]: Move this to the `## Search Hugging Face` section

File: examples/model_search/README.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+
+# Inpainting
+pipeline = EasyPipelineForInpainting.from_huggingface(
+    "search_word",
+    checkpoint_format="diffusers",
+).to("cuda")
+```
+
+
+### Application Examples
```
Comment (stevhliu) [inline examples/model_search/README.md:80]:
```suggestion
## Search Civitai and Hugging Face
```

Comment (stevhliu) [inline examples/model_search/README.md:80]:
```suggestion
## Search Civitai and Hugging Face
```

File: examples/model_search/README.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+| model_type      | string                 | `Checkpoint`  | The type of model to search for.  <br>(for example `Checkpoint`, `TextualInversion`, `Controlnet`, `LORA`, `Hypernetwork`, `AestheticGradient`, `Poses`)      |
+| base_model      | string                 | None          | Trained model tag (for example  `SD 1.5`, `SD 3.5`, `SDXL 1.0`) |
+| torch_dtype     | string, torch.dtype    | None          | Override the default `torch.dtype` and load the model with another dtype.     |
+| force_download  | bool                   | False         | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
+| cache_dir       | string, Path | None    | Path to the folder where cached files are stored. |
+| resume          | bool   | False         | Whether to resume an incomplete download. |
+| token           | string | None          | API token for Civitai authentication. |
+
+
+### `search_civitai` parameters
```
Comment (stevhliu) [inline examples/model_search/README.md:127]: Move to `## Search Civitai and Hugging Face`

File: examples/model_search/README.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+| search_word           | string, Path        | ー             | The search query string. Can be a keyword, Hugging Face URL, local directory or file path, or a Hugging Face path (`<creator>/<repo>`). |
+| checkpoint_format     | string              | `single_file`  | The format of the model checkpoint.<br>● `single_file` to search for `single file checkpoint` <br>●`diffusers` to search for `multifolder diffusers format checkpoint` |
+| torch_dtype           | string, torch.dtype | None           | Override the default `torch.dtype` and load the model with another dtype. |
+| force_download        | bool                | False          | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
+| cache_dir             | string, Path        | None           | Path to a directory where a downloaded pretrained model configuration is cached if the standard cache is not used.   |
+| token                 | string, bool        | None           | The token to use as HTTP bearer authorization for remote files.  |
+
+
+### `search_huggingface` parameters
```
Comment (stevhliu) [inline examples/model_search/README.md:155]: Move to `## Search Civitai and Hugging Face` section

**Review by stevhliu — COMMENTED** [review]
Thanks! Just a few more formatting things :)

File: examples/model_search/README.md (hunk 9 above)
Comment (suzukimain) [reply examples/model_search/README.md]: Corrected.

**Comment by suzukimain** [comment]
Fixed. Please confirm. cc:@stevhliu

**Review by stevhliu — APPROVED** [review]
Cool, thanks so much for your project 🤗

Can you run `make style` to fix the failing CI test?

**Comment by suzukimain** [comment]
hi, @stevhliu  Corrected

**Comment by suzukimain** [comment]
Corrected pipeline style.

**Pull request merged**